client/src/components/Workflow/WorkflowList.vue

```diff
@@ -143,47 +156,44 @@ export default {
             loading: true,
             message: null,
             messageVariant: null,
-            nWorkflows: 0,
-            workflows: [],
             titleSearchWorkflows: _l("Search Workflows"),
             titleCreate: _l("Create"),
             titleImport: _l("Import"),
             titleRunWorkflow: _l("Run workflow"),
+            workflowItemsModel: [],
+            workflowItems: [],
+            perPage: this.rowsPerPage(20),
         };
     },
     computed: {
         showNotFound() {
-            return this.nWorkflows === 0 && this.filter;
+            return this.filter;
         },
         showNotAvailable() {
-            return this.nWorkflows === 0 && !this.filter;
+            return !this.filter;
         },
         showMessage() {
             return !!this.message;
         },
+        apiUrl() {
+            return `${getAppRoot()}api/workflows`;
+        },
     },
     created() {
         this.root = getAppRoot();
         this.services = new Services({ root: this.root });
-        this.load();
     },
-    methods: {
-        load() {
-            this.loading = true;
-            this.filter = "";
-            this.services
-                .getWorkflows()
-                .then((workflows) => {
-                    this.workflows = workflows;
-                    this.nWorkflows = workflows.length;
-                    this.loading = false;
-                })
-                .catch((error) => {
-                    this.error = error;
-                });
+    watch: {
+        filter(val) {
+            this.refresh();
         },
-        filtered: function (items) {
-            this.nWorkflows = items.length;
+    },
+    methods: {
+        provider(ctx) {
+            ctx.apiUrl = this.apiUrl;
+            const extraParams = { search: this.filter, skip_step_counts: true };
+            this.workflowItems = storedWorkflowsProvider(ctx, this.setRows, extraParams);
```

Comment:
This needs an `await` ;)
```suggestion
        async provider(ctx) {
            ctx.apiUrl = this.apiUrl;
            const extraParams = { search: this.filter, skip_step_counts: true };
            this.workflowItems = await storedWorkflowsProvider(ctx, this.setRows, extraParams);
```